java/src/org/openqa/selenium/grid/sessionmap/config/SessionMapOptions.java

```diff
@@ -78,6 +81,15 @@ public URI getSessionMapUri() {
     }
   }
 
+  public boolean isSecure() {
+    return config.get(SERVER_SECTION, "https-private-key").isPresent()
+        && config.get(SERVER_SECTION, "https-certificate").isPresent();
+  }
+
+  public boolean isSelfSigned() {
+    return config.getBool(SERVER_SECTION, "https-self-signed").orElse(false);
+  }
+
```

Comment:
This could be implemented in `BaseServerOptions` and reuse across other classes.